app/models/visit.rb

```diff
@@ -28,6 +28,16 @@ def total_number_of_visitors
     :allowance_will_renew?, :allowance_renews_on,
     to: :rejection
 
+  scope :by_day, -> {
+    group('id, extract(day from created_at::timestamp::date)::integer')
+  }
+  scope :by_week, -> {
+    group('id, extract(week from created_at::timestamp::date)::integer')
```

Comment:
This needs to include the `isoyear` to be accurate for more than the first year.